src/VirtualTree.lua

```diff
@@ -0,0 +1,33 @@
+local ShallowWrapper = require(script.Parent.ShallowWrapper)
+local Type = require(script.Parent.Type)
+
+local config = require(script.Parent.GlobalConfig).get()
+
+local VirtualTree = {}
+
+function VirtualTree.new(rootNode, internalDataSymbol)
+	local tree = {
+		[Type] = Type.VirtualTree,
+		[internalDataSymbol] = {
+			rootNode = rootNode,
+			mounted = true,
+		}
+	}
+
+	function tree:getTestRenderOutput(options)
```

Comment:
for a V1 we might want to hide shallows as an implementation detail. We can have a single API that will be really easy to use in tests, for example:

`
local tree = Roact.mount(element)
Roact.matchSnapshot(tree)
Roact.unmount(tree)`